src/java/org/apache/cassandra/net/InboundConnectionInitiator.java

```diff
@@ -98,17 +95,22 @@ public void initChannel(SocketChannel channel) throws Exception
 
             // order of handlers: ssl -> logger -> handshakeHandler
             // For either unencrypted or transitional modes, allow Ssl optionally.
-            if (settings.encryption.optional)
+            switch(settings.encryption.tlsEncryptionPolicy())
             {
-                pipeline.addFirst("ssl", new OptionalSslHandler(settings.encryption));
-            }
-            else
-            {
-                SslContext sslContext = SSLFactory.getOrCreateSslContext(settings.encryption, true, SSLFactory.SocketType.SERVER);
-                InetSocketAddress peer = settings.encryption.require_endpoint_verification ? channel.remoteAddress() : null;
-                SslHandler sslHandler = newSslHandler(channel, sslContext, peer);
-                logger.trace("creating inbound netty SslContext: context={}, engine={}", sslContext.getClass().getName(), sslHandler.engine().getClass().getName());
-                pipeline.addFirst("ssl", sslHandler);
+                case UNENCRYPTED:
+                    // Handler checks for SSL connection attempts and cleanly rejects them if encryption is disabled
+                    pipeline.addFirst("rejectssl", new rejectSslHandler());
+                    break;
+                case OPTIONAL:
+                    pipeline.addFirst("ssl", new OptionalSslHandler(settings.encryption));
+                    break;
+                case ENCRYPTED:
+                    SslContext sslContext = SSLFactory.getOrCreateSslContext(settings.encryption, true, SSLFactory.SocketType.SERVER);
```

Comment:
Nit: If its not a big deal, could you create a constant `BUILD_TRUST_STORE=true` and pass it in. Makes the code a bit more readable. If you feel too ambitious, make this into a builder.

Reply:
Went with the low effort route as there's already more than I wanted to refactor in the beta.